patching/types/src/index.ts

```diff
@@ -3,7 +3,25 @@ export interface PatchFile {
   hash: string
 }
 
+// TODO: replace all occurrences of PatchInfo with PatchFile before the next major version is released
```

Comment:
After merging the PR put any pending tasks for v11 to the related discussion: https://github.com/orgs/pnpm/discussions/9238

Reply:
Maybe v11 should remove both `allowNonAppliedPatches` and `strictPatches` completely? Patch application would always be strict.

Reply:
I think `allowNonAppliedPatches` should be kept because what if an org has a shareable set of patches that are added to all projects. It is fine if some of these patches are not used.

Reply:
Then I think it would make sense to rename it to `allowUnusedPatch` now. `allowNonAppliedPatch` should still be removed. But if that's the case, perhaps we should also replace `strictPatches` with `allowPatchFailure` (which would no longer manage unused patches).

Reply:
it could be `ignorePatchFailure` and `allowUnusedPatch`.